vyper/semantics/analysis/base.py

```diff
@@ -234,6 +234,18 @@ class VarAccess:
     # A sentinel indicating a subscript access
     SUBSCRIPT_ACCESS: ClassVar[Any] = object()
 
+    def __reduce__(self):
+        dict_obj = {f.name: getattr(self, f.name) for f in fields(self)}
+        return self.__class__._produce, (dict_obj,)
+
+    @classmethod
+    def _produce(cls, data):
+        ret = cls.__new__(cls)
+        # bypass blocking of setattr by `frozen=True`
```

Comment:
should we use `frozen` instances if we use this bypass?

Reply:
Yes, this is only during a specialized instance of object construction to work around a bug in pickle. The fields cannot be modified after construction.

Reply:
See for instance how the default dataclasses setstate works:
https://github.com/python/cpython/blob/16cd6cc86b3ba20074ae3eefb61aeb24ee1544f7/Lib/dataclasses.py#L1198

Reply:
Why not simply return `cls(**data)`?

Reply:
good idea, thanks!
d3309e75ca12500d979bbc15f22808bea7f52c12